src/main/java/org/threadly/concurrent/PrioritySchedulerStatisticTracker.java

```diff
@@ -769,50 +606,207 @@ private static double getTruePercent(Collection<Boolean> list) {
   }
   
   /**
-   * Called at the start of execution to track statistics around task execution.
+   * <p>This class primarily holds the structures used to store the statistics.  These can not be 
+   * maintained in the parent class since sub classes need to be able to access them.  This is to 
+   * help facilitate allowing the parent class to be garbage collected freely despite references 
+   * to subclasses from other thread garbage collection roots.</p>
    * 
-   * @param taskWrapper Wrapper that is about to be executed
-   */
-  protected void trackTaskStart(Wrapper taskWrapper) {
-    runningTasks.put(taskWrapper, taskWrapper.startTime = Clock.accurateForwardProgressingMillis());
+   * @author jent - Mike Jensen
+   * @since 3.5.0
+   */
+  protected static class StatsManager {
```

Comment:
This class holds the shared memory for statistics collections.  It is passed into the wrappers of tasks to be executed, as well as the overriding version of WorkerPool.